packages/query/src/block-processor.ts

```diff
@@ -162,12 +162,12 @@ export class BlockProcessor implements BlockProcessorInterface {
 
     // special case genesis sync
     if (currentHeight === -1n) {
```

Comment:
question: what is the difference between this conditional and the one nested inside?

```
if (currentHeight === -1n) {
vs
if (this.genesisBlock?.height === currentHeight + 1n) {
```

Reply:
issue: Ah just realized the second conditional only runs if there is a bundled genesis block for the chain. Can you add a code comment?

Reply:
added more information on the conditionals